Source/Dafny/Rewriter.cs

```diff
@@ -1361,15 +1361,14 @@ public AutoReqFunctionRewriter(ErrorReporter reporter)
           e.UpdateTerm(allReqsSatisfiedAndTerm);
           reporter.Info(MessageSource.Rewriter, e.tok, "autoreq added (" + Printer.ExtendedExprToString(allReqsSatisfied) + ") &&");
         }
-      } else if (expr is SetComprehension) {
-        var e = (SetComprehension)expr;
+      } else if (expr is SetComprehension setComprehension) {
```

Comment:
I'll do the change.

Reply:
So I tried to put "e" instead of "setComprehension", but now it generate compile errors, because unfortunately the following is invalid csharp:

```
if(expr is LetExpr e) {
} else if(expr is ThisExpr e) {
}
```
It says that the e conflict because they are declared in the same scope. I agree that this is weird and unexpected.

Now I have the choice between:
1) Revert back where I was before trying to implement your change.
2) Implement my change in every guard so that it is consistent, albeit with different variable names
3) Reinject the inner `var e` declaration as before.
4) Use of a switch statements (which tolerates same variable names) but then we need to include break statements at the end of every case.

Everything said, I believe that 2) would be the best for now. First, it did not bother me not knowing that `expr` and `setComprehension` were the same (it's a CTRL+Click away). Moreover, since the name of the variable `setComprehension` is explicit, I know at every step what I was manipulating in different branches, and when debugging.
What do you prefer?

Reply:
I misread your comment, it looks like you were already aware of all of this.
I ended up doing it the old way.